Python/Product/VSCode/AnalysisVsc/LanguageServer.cs

```diff
@@ -69,51 +89,65 @@ struct PublishDiagnosticsParams {
         private void OnLogMessage(object sender, LogMessageEventArgs e) => _ui.LogMessage(e.message, e.type);
         private void OnPublishDiagnostics(object sender, PublishDiagnosticsEventArgs e) {
             var parameters = new PublishDiagnosticsParams {
-                uri = e.uri,
+                uri = e.uri.ToString(),
                 diagnostics = e.diagnostics.ToArray()
             };
-            _rpc.InvokeAsync("textDocument/publishDiagnostics", parameters);
+            _vscode.NotifyWithParameterObjectAsync("textDocument/publishDiagnostics", parameters);
         }
-        private void OnApplyWorkspaceEdit(object sender, ApplyWorkspaceEditEventArgs e) => _rpc.InvokeAsync("workspace/applyEdit", e.@params);
-        private void OnRegisterCapability(object sender, RegisterCapabilityEventArgs e) => _rpc.InvokeAsync("client/registerCapability", e.@params);
-        private void OnUnregisterCapability(object sender, UnregisterCapabilityEventArgs e) => _rpc.InvokeAsync("client/unregisterCapability", e.@params);
+        private void OnApplyWorkspaceEdit(object sender, ApplyWorkspaceEditEventArgs e)
+            => _vscode.NotifyWithParameterObjectAsync("workspace/applyEdit", e.@params);
+        private void OnRegisterCapability(object sender, RegisterCapabilityEventArgs e)
+            => _vscode.NotifyWithParameterObjectAsync("client/registerCapability", e.@params);
+        private void OnUnregisterCapability(object sender, UnregisterCapabilityEventArgs e)
+            => _vscode.NotifyWithParameterObjectAsync("client/unregisterCapability", e.@params);
         #endregion
 
         #region Lifetime
         [JsonRpcMethod("initialize")]
         public Task<InitializeResult> Initialize(JToken token) {
             var p = token.ToObject<InitializeParams>();
+            _ui.LogMessage($"VS Code process: {(p.processId.HasValue ? p.processId.Value : 0)}", MessageType.Info);
+
+#if WAIT_FOR_DEBUGGER
+            while (!System.Diagnostics.Debugger.IsAttached) {
+                Thread.Sleep(1000);
```

Comment:
`await Task.Delay(1000)`

Reply:
It's current in Program.cs which is not async